libvmaf/include/libvmaf/libvmaf.h

```diff
@@ -63,25 +63,28 @@ enum VmafPoolingMethod {
  *
  * @param n_threads   How many threads can be used to run
  *                    feature extractors concurrently.
- * 
+ *
  * @param n_subsample Compute scores only every N frames.
  *                    Note that setting an even value for N can lead to
  *                    inaccurate results. For more detail, see
  *                    https://github.com/Netflix/vmaf/issues/1214
- * 
+ *
  * @param cpumask     Restrict permitted CPU instruction sets.
  *                    if cpumask & 1:  disable SSE2 / disable NEON (on arm64)
  *                    if cpumask & 2:  disable SSE3/SSSE3
  *                    if cpumask & 4:  disable SSE4.1
  *                    if cpumask & 8:  disable AVX2
  *                    if cpumask & 16: disable AVX512
  *                    if cpumask & 32: disable AVX512ICL
- * 
+ *
  * @param gpumask     Restrict permitted GPU operations.
  *                    if gpumask: disable CUDA
  */
 typedef struct VmafConfiguration {
     enum VmafLogLevel log_level;
+    void (*callback)(void*, const char *, double);
+    void* meta_data;
+    size_t meta_data_sz;
```

Comment:
Rather than configure this with  `VmafConfiguration` could we please introduce another api function to register these callbacks? The function should take an already allocated VmafContext and it's own configuration struct. If there is a use case for registering more than just a single callback, we should be able to support that by calling this new api more than once.

Reply:
Should a new function automatically unregister existing callback when attempting to register itself, or should we define a new function for unregistering callbacks?

Reply:
When we register the callback, the `VmafContext` should take ownership of the callback/associated memory, when the `VmafContext` is destroyed, anything related to the callback(s) can be cleaned up. The ownership of data passed into the callback I think still belongs with the user. Thoughts?

Reply:
Thank you for the explanation! This approach looks logical and helps maintain responsibility boundaries.

I am planning a to use a linked list structure something like this;
```c
typedef struct VmafPropagateMetadataNode {
       void (*callback)(void *, const char *, double);
       void* meta_data;
       size_t meta_data_sz;
       VmafPropagateMetadataNode *next;
}
```

With that we could support multiple callbacks as you told, What do you think?

And could you say what do you think about the parameters of callback?

```c
void (*callback)(void*, const char *, double);
```

I can take an offset value instead of `const char *` and `double` and I can change with `offset` value. However current one looks more sensible to me.